protocols/smtp/src/test/java/org/apache/james/protocols/smtp/utils/TestMessageHook.java

```diff
@@ -18,37 +18,40 @@
  ****************************************************************/
 package org.apache.james.protocols.smtp.utils;
 
-import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
 
-import org.apache.commons.configuration.Configuration;
-import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.ConfigurationException;
 import org.apache.james.protocols.smtp.MailEnvelope;
 import org.apache.james.protocols.smtp.SMTPSession;
 import org.apache.james.protocols.smtp.hook.HookResult;
 import org.apache.james.protocols.smtp.hook.HookReturnCode;
 import org.apache.james.protocols.smtp.hook.MessageHook;
 
+import com.google.common.collect.ImmutableList;
+
 public class TestMessageHook implements MessageHook {
 
-    private final List<MailEnvelope> queued = new ArrayList<MailEnvelope>();
+    private final Collection<MailEnvelope> queued = new ConcurrentLinkedQueue<MailEnvelope>();
     
     public HookResult onMessage(SMTPSession session, MailEnvelope mail) {
         queued.add(mail);
         return new HookResult(HookReturnCode.OK);
     }
  
     public List<MailEnvelope> getQueued() {
-        return queued;
+        return ImmutableList.copyOf(queued);

```

Comment:
I would have use http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CopyOnWriteArrayList.html , doesn't it fit better ?